client-runtime/src/main/java/com/microsoft/rest/policy/LoggingPolicy.java

```diff
@@ -101,21 +103,21 @@ private void log(Logger logger, String s) {
         }
 
         final long startNs = System.nanoTime();
```

Comment:
I think you're right..otherwise, it's timing from when the HttpRequest was created, plus any time spent sitting around not subscribing, plus any time taken for the request to go out on the wire and for the response to be read from the wire.

Interesting that in the case of response bodies, this might mean that tookMs is just considering the time needed to read the response status and headers, not the body. And you don't know how long it took to read the body unless you add a doOnSubscribe to the body content--but you don't know if anyone WILL subscribe to the body content. :joy:

Reply:
right :laughing: